packages/x-components/tests/e2e/cucumber/multiselect-filters.spec.ts

```diff
@@ -1,46 +1,24 @@
 import { And, Then, When } from 'cypress-cucumber-preprocessor/steps';
 
-/**
- * Click on a filter from a certain facet.
- *
- * @param facetName - Name of the facet which the filter to be clicked belongs to.
- * @param nthFilter - Position of the filter to be clicked.
- */
-function clickFacetNthFilter(facetName: string, nthFilter: number): void {
-  cy.getByDataTest(`${facetName}-filter`).eq(nthFilter).click().invoke('text').as('clickedFilter');
-}
-
-When(
-  'filter number {int} is selected in facet {string}',
-  (filterNumber: number, facetName: string) => {
-    clickFacetNthFilter(facetName, filterNumber);
-  }
-);
+When('filter {string} is clicked in facet {string}', (filter: string, facetName: string) => {
+  cy.getByDataTest(`${facetName}-filter`).contains(filter).click();
+});
 
 And('waiting for search request intercept', () => {
   cy.intercept('https://api.empathy.co/search').as('requestWithFilter');
 });
 
-Then(
-  'selected filter is shown in the selected filters list',
-  function (this: { clickedFilter: string }) {
-    cy.getByDataTest('selected-filters-list')
-      .getByDataTest('selected-filters-list-item')
-      .getByDataTest('filter')
-      .invoke('text')
-      .then(filterName => {
-        expect(filterName).to.eq(this.clickedFilter);
-      });
-  }
-);
+Then('filter {string} is shown in the selected filters list', (filter: string) => {
+  cy.getByDataTest('selected-filters-list')
+    .getByDataTest('filter')
+    .contains(filter)
+    .should('exist');
+});
 
-And('search request contains selected filter', function (this: { clickedFilter: string }) {
-  cy.wait('@requestWithFilter').its('request.body').should('include', this.clickedFilter);
+And('search request contains {string} filter', (filter: string) => {
+  cy.wait('@requestWithFilter').its('request.body').should('include', filter);
```

Comment:
Should we use here the `json.Parse` as the other PR, I really like it.

Reply:
I think they are both different cases. Here it would be more complicated because it is searching a filter by its label. I would leave it like this.